core/src/main/java/org/elasticsearch/action/search/SearchTransportService.java

```diff
@@ -193,6 +194,13 @@ public RemoteClusterService getRemoteClusterService() {
         return transportService.getRemoteClusterService();
     }
 
+    /**
+     * Return a map of nodeId to pending number of requests for the given action name prefix
+     */
+    public Map<String, Long> getPendingRequests(final String actionNamePrefix) {
```

Comment:
Looks better to me but I think I'd rather like a Map<String, AtomicLong>, ideally pre-filled with every possible action name so that the map is effectively immutable afterwards and concurrency is only handled at the AtomicLong level? It would also create fewer boxed longs.